### ISSUE TITLE
AJAX

### ISSUE THREAD

**Issue by haxwell** [haxwell/quizki#58]
The application needs to be more responsive.. we need to do without submitting the page wherever possible.. So, in short we need AJAX..

to do it, it will take a pretty major rewrite of how we do things now.. we are using objects from the Session and processing them using JSTL now.. but we will need to use Javascript to process arrays in order to make use of Javascript.. 

For instance, in profile-questions.jsp, I'd like to be able to press a key (and then after a delay, to see if I press another key before that), and then call the server with that string and have it return an array of elements related to questions, which can then be quickly populated in the UI table.. this should work for all the filters (send filter type and string, get back JS array to populate the UI). 

So, to do that we'll need to change the way tables are populated, so that they all use JS arrays instead of Java Objects. So, profile-questions, profile-exams, generateExams, exam.jsp and questions.jsp (at least) will need that type of change..

Then similar changes necessary in takeExam.. 

In short lots of changes coming.. AJAX will be major..
